0001-typedb/02-dev/01-connections.md

```diff
@@ -0,0 +1,371 @@
+
+// get database schema
+client.databases().get("test-db").schema();
+
+// get all databases
+client.databases().all();
+
+// check if database exists
+client.databases().contains("test-db");
+
+// delete database
+client.databases().get("test-db").delete();
+```
+
+[tab:end]
+
+[tab:Python]
+
+<!-- test-ignore -->
+```python
+# create database
+client.databases().create("test-db")
+
+# get database schema
+client.databases().get("test-db").schema()
+
+# get all databases
+client.databases().all()
+
+# check if database exists
+client.databases().contains("test-db")
+
+# delete database
+client.databases().get("test-db").delete()
+```
+
+[tab:end]
+</div>
+
+## Sessions
+
+There are two types of sessions: 
+
+- SCHEMA sessions,
+- DATA sessions. 
+
+| Session type | Read data | Write data | Read schema  | Write schema |
+|--------------|-----------|------------|--------------|--------------|
+| DATA         | Yes       | Yes        | Yes          | Yes          |
+| SCHEMA       | Yes       | No         | Yes          | No           |
+
+TypeDB clients should read and write data in DATA sessions.
+
+TypeDB clients should read and write schema in SCHEMA sessions.
+
+<div class="note">
+[Note]
+If a client needs to read both schema and data from a database, it can be done in any session type. But it is 
+NOT possible to modify a schema and its data in the same session, regardless of the type. Write transactions are strict 
+to the session types.
+</div>
+
+Once a session has been opened, clients can open and close transactions to read and write a database’s schema or data.
+
+<div class="tabs dark">
+
+[tab:TypeDB Console]
+
+<!-- test-ignore -->
+```
+transaction iam data read
+```
+
+[tab:end]
+
+[tab:Java]
+
+<!-- test-ignore -->
+```java
+TypeDBSession session = client.session("iam", TypeDBSession.Type.DATA);
+```
+
+[tab:end]
+
+[tab:Node.js]
+
+<!-- test-ignore -->
+```javascript
+session = await client.session("iam", SessionType.DATA);
+```
+
+[tab:end]
+
+[tab:Python]
+
+<!-- test-ignore -->
+```python
+session = client.session("iam", SessionType.DATA)
+```
+
+[tab:end]
+</div>
+
+<div class="note">
+[Note]
+It is recommended to avoid long-running sessions, because of possible network failures.
+</div>
+
+A good principle to follow for logically coherent transactions to be grouped together into a session.
+
+## Transactions
+
+All queries to a TypeDB database are performed through transactions. TypeDB transactions provide full 
+[ACID guarantees](#acid-guarantees) up to [snapshot isolation](#isolation).
+
+There are two types of transactions: 
+
+- READ transactions
+- WRITE transactions
+
+In addition, transactions must be explicit — clients must open and close transactions via an API.
+
+<div class="tabs dark">
+
+[tab:TypeDB Console]
+
+<!-- test-ignore -->
+```typeql
+# start transaction
+transaction iam data write
+
+insert $x isa person;
+$x has full-name "Kevin";
+$x has email "Kevin@vaticle.com";
+
+# commit changes
+commit
```

Comment:
No, right now we can't. We have only input in the code block. And no output. Mostly so that the commands can be copied easily. Maybe sometime later we will have a feature to show some output that can't be copied by mistake, but not now.